src/server/game/Entities/AreaTrigger/AreaTrigger.cpp

```diff
@@ -141,6 +168,16 @@ void AreaTrigger::Update(uint32 p_time)
     WorldObject::Update(p_time);
     _timeSinceCreated += p_time;
 
+    if (GetTemplate()->HasFlag(AREATRIGGER_FLAG_HAS_ATTACHED))
+    {
+        if (Unit* caster = GetCaster())
+        {
+            Relocate(caster);
```

Comment:
Relocate does not trigger visibility updates, you need to implement something like Map::CreatureRelocation